tool/teleport/common/configurator.go

```diff
@@ -126,14 +139,24 @@ func onConfigureDiscoveryBootstrap(flags configureDiscoveryBootstrapFlags) error
 		return trace.Wrap(err)
 	}
 
-	fmt.Printf("Reading configuration at %q...\n\n", flags.config.ConfigPath)
+	if flags.config.DiscoveryService && flags.databaseServiceRole != "" {
+		config := makeDatabaseServiceBootstrapFlagsWithDiscoveryServiceConfig(flags)
```

Comment:
The logic here is a bit confusing: this clause is protected in part by `if flags.config.DiscoveryService`, yet we're calling a function which returns a `config` with `config.DiscoveryService = false`. Can you add a comment that explains what's going on here?